sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -3646,3 +3649,344 @@ def hpat_pandas_series_pct_change_impl(self, periods=1, fill_method='pad', limit
+    -------
+    :obj:`pandas.Series`
+        The result of the operation
+    """
+
+    _func_name = 'Operator add().'
+
+    ty_checker = TypeChecker('Operator add().')
+    ty_checker.check(self, SeriesType)
+
+    if not isinstance(other, (SeriesType, types.Number)):
+        ty_checker.raise_exc(other, 'pandas.series or scalar', 'other')
+
+    series_indexes_alignable = False
+    if isinstance(other, SeriesType):
+        if (other.index == string_array_type and self.index == string_array_type):
+            series_indexes_alignable = True
+
+        if ((isinstance(self.index, types.NoneType) or
+             isinstance(self.index, types.Array) and isinstance(self.index.dtype, types.Number))
+            and (isinstance(other.index, types.NoneType) or
+                 isinstance(other.index, types.Array) and isinstance(other.index.dtype, types.Number))):
+            series_indexes_alignable = True
+
+    if isinstance(other, SeriesType) and not series_indexes_alignable:
+        raise TypingError('{} Not implemented for series with not-alignable indexes. \
+        Given: self.index={}, other.index={}'.format(_func_name, self.index, other.index))
+
+    # specializations for numeric series - TODO: support arithmetic operation on StringArrays
+    if (isinstance(other, types.Number)):
+        def hpat_pandas_series_add_scalar_impl(self, other):
+            return pandas.Series(self._data + other, self._index)
+
+        return hpat_pandas_series_add_scalar_impl
+
+    elif (isinstance(other, SeriesType)):
+        is_numeric_index = isinstance(self.index, (types.Array, types.NoneType))
+
+        if is_numeric_index:
+            ty_left_index_dtype = types.int64 if isinstance(self.index, types.NoneType) else self.index.dtype
+            ty_right_index_dtype = types.int64 if isinstance(other.index, types.NoneType) else other.index.dtype
+            np_index_dtypes = [numpy_support.as_dtype(ty_left_index_dtype),
+                               numpy_support.as_dtype(ty_right_index_dtype)]
+            np_index_common_dtype = numpy.find_common_type([], np_index_dtypes)
+            numba_index_common_dtype = numpy_support.from_dtype(np_index_common_dtype)
+
+        if (isinstance(self.index, types.NoneType) and isinstance(other.index, types.NoneType)):
+            def hpat_pandas_series_add_impl(self, other):
+
+                if (len(self._data) == len(other._data)):
+                    return pandas.Series(numpy.asarray(self._data + other._data, numpy.float64))
+                else:
+                    min_data_size = min(len(self._data), len(other._data))
+                    max_data_size = max(len(self._data), len(other._data))
+                    new_data = numpy.empty(max_data_size, dtype=numpy.float64)
+                    new_data[:min_data_size] = self._data[:min_data_size] + other._data[:min_data_size]
+                    new_data[min_data_size:] = numpy.repeat(numpy.nan, max_data_size - min_data_size)
```

Comment:
This behavior could be changed by parameters passed to `add` method of series.

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.add.html#pandas.Series.add

This PR is about '+' operator, but what about 'add' method?

Reply:
@AlexanderKalistratov Yes, I understand, I actually planned to have two different templates: one for binary operators and the other for add/sub/div/... methods, which can then be used with auto-generation script, because latter need special handling for method args and so on. But yes, I will address that in subsequent PRs.